### PR TITLE
Add a minimal-dependencies working example listening to k8s events

### LINKED ISSUE #203: Migrate all the samples from eventing/sample to docs/eventing/sample

### PR DESCRIPTION
**Pull request by evankanderson**
Fixes #203 

## Proposed Changes

* Adds a sample which uses Feed and Service to listen to k8s events.
* Documents using some of the released eventing components (still TODO on how to document the release, probably in `eventing/README.md`).
* Uses `dep` to vendor the dependencies of `function.go` for the moment. I'll send a PR to remove the `glog` usage from https://github.com/knative/eventing/tree/master/pkg/event later today, which may remove the need to use `dep` to avoid double declaration of the `log_dir` flag.
* Ran through the sample twice, but may have missed bugs or assumptions.

### CODE REVIEW

**Comment by evankanderson** [comment]
For your reading pleasure (images work):

https://github.com/evankanderson/docs/tree/1-10th-event/eventing/samples/k8s-events

**Comment by evankanderson** [comment]
Let me see if I can excise the other commits that somehow leaked into my repo.

File: eventing/samples/k8s-events/README.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+## Prerequisites
+
+- A Kubernetes cluster with Knative installed. Follow the
+  [installation instructions](https://github.com/knative/docs/blob/master/install/README.md)
+  if you need to create one.
+- [Docker](https://www.docker.com/) installed and running on your local machine,
+  and a Docker Hub account configured (you'll use it for a container registry).
+- The core Knative eventing tools installed. You can install them with:
+  ```shell
+  kubectl apply -f https://storage.googleapis.com/knative-releases/latest/release-eventing.yaml
```
Comment (grantr) [inline eventing/samples/k8s-events/README.md]: This url is now `https://storage.googleapis.com/knative-releases/eventing/latest/release.yaml`

File: eventing/samples/k8s-events/README.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+  ```
+
+## Configuring Knative
+
+To use this sample, you'll need to install the `stub` ClusterBus and the
+`k8sevents` EventSource.
+
+```shell
+kubectl apply -f https://storage.googleapis.com/knative-releases/latest/release-eventing-clusterbus-stub.yaml
```
Comment (grantr) [inline eventing/samples/k8s-events/README.md]: This url is now `https://storage.googleapis.com/knative-releases/eventing/latest/release-clusterbus-stub.yaml`

File: eventing/samples/k8s-events/README.md [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+kubectl apply -f https://storage.googleapis.com/knative-releases/latest/release-eventing-source-k8sevents.yaml
```
Comment (grantr) [inline eventing/samples/k8s-events/README.md]: This url is now `https://storage.googleapis.com/knative-releases/eventing/latest/release-source-k8sevents.yaml`

File: eventing/samples/k8s-events/README.md [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+access of this service account to only specific namespaces.
+
+```shell
+kubectl apply -f serviceaccount.yaml
+```
+
+## Build and deploy the sample
+
+1.  Use Docker to build the sample code into a container. To build and push with
+    Docker Hub, run these commands replacing {username} with your Docker Hub
```
Comment (grantr) [inline eventing/samples/k8s-events/README.md]: Add backticks around `{username}`

File: eventing/samples/k8s-events/README.md [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+    # Build the container on your local machine
+    docker build -t {username}/k8s-events --file=eventing/samples/k8s-events/Dockerfile .
+
+    # Push the container to docker registry
+    docker push {username}/k8s-events
+    ```
+
+1.  After the build has completed and the container is pushed to Docker Hub, you
+    can deploy the function into your cluster. **Ensure that the container image
+    value in function.yaml matches the container you built in the previous step.**
```
Comment (grantr) [inline eventing/samples/k8s-events/README.md]: Add backticks around `function.yaml`. Maybe add an example of the image value?

File: eventing/samples/k8s-events/README.md [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+    read-k8s-events   True      <none>
+    ```
+
+1.  Create the flow sending Kubernetes Events to the service:
+
+    ```shell
+    kubectl apply -f eventing/samples/k8s-events/flow.yaml
+    ```
+
+1.  You can read the function logs using Kibana. To access Kibana, you need to
```
Comment (grantr) [inline eventing/samples/k8s-events/README.md]: Might be worth a footnote mentioning that Kibana isn't installed in the lite or no-mon installs and that`kubectl logs` is an alternative.

File: eventing/samples/k8s-events/function.yaml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+metadata:
+  name: read-k8s-events
+  namespace: default
+spec:
+  runLatest:
+    configuration:
+      revisionTemplate:
+        spec:
+          container:
+            # Replace this image with your {username}/k8s-events container
```
Comment (grantr) [inline eventing/samples/k8s-events/function.yaml]: Maybe add a `TODO` or some other callout to be clear that this requires action from the user.

File: eventing/samples/k8s-events/flow.yaml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+spec:
+  serviceAccountName: feed-sa
+  trigger:
+    eventType: dev.knative.k8s.event
+    # This is kind of superfluous due to parameters, but the current
+    # feed model still requires a resource
+    # Future revisions may opt to have a meaningful resource string
+    # that allows users to scope which objects they choose based on
+    # a templatized self-link. E.g. Configurations across all namespaces:
+    # /apis/serving.knative.dev/v1alpha1/namespaces/{namespace}/configurations/{config}
```
Comment (grantr) [inline eventing/samples/k8s-events/flow.yaml]: I'd probably remove all of these comment lines. They're distracting and won't have meaning to the new user.

**Pull request merged**